### PR TITLE
Fixes #742 - Added slf4j implementation to tests

### LINKED ISSUE #742: Add slf4j implementation to pom.xml for tests
See discussion in #741

### PR DESCRIPTION
**Pull request by mikewalch**
* Added slf4j-log4j12 to all modules with tests (in test scope)
* Removed any direct dependencies on logj4 which is now brought in by slf4j-log4j12
* Removed reference to fluo-metrics (old non-existent module) from project pom.
* Made dependency analysis plugin ignore slf4j-log4j12 and log4j.
* Removed log4j-over-slf4j from project pom as it's no longer used.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -343,11 +333,13 @@
             <configuration>
               <failOnWarning>true</failOnWarning>
               <ignoredDependencies>
+                <ignoredDependency>log4j:log4j:jar:*</ignoredDependency>
```
Comment (keith-turner) [inline pom.xml]: why was this needed?

Comment (mikewalch) [inline pom.xml]: log4j is used in test code to change log levels.  however, it is brought in by slf4-logj412 so it does not need to explicitly declared.

**Comment by keith-turner** [comment]
+1

File: pom.xml (hunk 1 above)
Comment (ctubbsii) [inline pom.xml]: It should be explicitly declared in the test scope if its classes are used in the test scope.
